plasmapy/formulary/parameters.py

```diff
@@ -166,77 +204,90 @@ def Alfven_speed(
     density: [u.m ** -3, u.kg / u.m ** 3],
     ion: Optional[Particle] = None,
-    z_mean=None,
+    z_mean: Optional[numbers.Real] = None,
 ) -> u.m / u.s:
     r"""
-    Return the Alfvén speed.
+    Calculate the Alfvén speed.
 
-    **Aliases:** `va_`
+    The Alfven speed :math:`V_A` is the typical propagation speed of magnetic
+    disturbances in a plasma, and is given by:
 
-    Parameters
-    ----------
-    B : ~astropy.units.Quantity
-        The magnetic field magnitude in units convertible to tesla.
+    .. math::
 
-    density : ~astropy.units.Quantity
-        Either the ion number density in units convertible to 1 / m**3,
-        or the mass density in units convertible to kg / m**3.
+        V_A = \frac{B}{\sqrt{\mu_0\rho}}
 
-    ion : ~plasmapy.particles.Particle, optional
-        Representation of the ion species (e.g., `'p'` for protons,
-        `'D+'` for deuterium, or `'He-4 +1'` for singly ionized
-        helium-4). If no charge state information is provided, then the
-        ions are assumed to be singly charged. If the density is an ion
-        number density, then this paramter is required in order to convert
-        to mass density.
+    where :math:`B` is the magnetic field and :math:`\rho = n_i m_i + n_e m_e`
+    is the total mass density (:math:`n_i` is the ion number denisty,
+    :math:`n_e` is the electron number denisty, :math:`m_i` is the ion mass,
+    and :math:`m_e` is the electron mass).
 
-    z_mean : ~astropy.units.Quantity, optional
-        The average ionization (arithmetic mean) for a plasma where the
-        a macroscopic description is valid. If this quantity is not
-        given then the atomic charge state (integer) of the ion
-        is used. This is effectively an average Alfven speed for the
-        plasma where multiple charge states are present.
+    **Aliases:** `va_`
+
+    Parameters
+    ----------
+    B : `~astropy.units.Quantity`
+        The magnetic field magnitude in units convertible to :math:`Tesla`.
+
+    density : `~astropy.units.Quantity`
+        Either the ion number density :math:`n_i` in units convertible to
+        :math:`m^{-3}` or the total mass density :math:`\rho` in units
+        convertible to :math:`kg / m^3`.
+
+    ion : `~plasmapy.particles.Particle`, optional
+        Representation of the ion species (e.g., `'p'` for protons, `'D+'` for
+        deuterium, `'He-4 +1'` for singly ionized helium-4, etc.). If no charge
+        state information is provided, then the ions are assumed to be singly
+        ionionized. If the density is an ion number density, then this paramter
+        is required in order to convert to mass density.
+
+    z_mean : `~numbers.Real`, optional
+        The average ionization state (arithmetic mean) of the ``ion`` composing
+        the plasma.  This is used in calculating the mass density
+        :math:`\rho = n_i (m_i + Z_{mean} m_e)`.  ``z_mean`` is ignored if
+        ``density`` is passed as a mass density.
```

Comment:
One use case I'm wondering about is if `ion` is provided (e.g., `He-4 2+`) and `z_mean` is provided as `1`.  In this case there's a contradiction between `ion` and `z_mean`...at least if we want to assume quasineutrality, which is usually the case.  (If I remember correctly, the derivation for the Alfvén speed implicitly assumes quasineutrality.)  And if `z_mean` is provided, then `ion` would be an `element` instead so the name `ion` wouldn't be appropriate. I'm wondering about using the machinery of `CustomParticle` for situations like this, which probably won't arise particularly frequently.  Hm...

Reply:
Yes, the derivation of the Alfvén speed assumes quasi-neutrality.  Now, if someone where to pass `ion="He-4 +2"` and `z_mean=1.2`, then the function will used the mass of `He-4 +2` to calculate the ion mass density and `z_mean=1.2` to calculate the electron mass density.  So, `z_mean` will take precedence over any charge value indicated by `ion`.

As I mentioned in the element chat, I'm not sure having `z_mean` as a `Particle` attribute is the right approach because `z_mean` is really a plasma attribute.